src/input/return_enums.rs

```diff
@@ -64,9 +64,16 @@ fn validate_iter_to_vec<'a, 's>(
         match validator.validate(py, item, extra, slots, recursion_guard) {
             Ok(item) => output.push(item),
             Err(ValError::LineErrors(line_errors)) => {
+                if !extra.exhaustive {
+                    return Err(ValError::Omit);
+                }
                 errors.extend(line_errors.into_iter().map(|err| err.with_outer_location(index.into())));
             }
-            Err(ValError::Omit) => (),
+            Err(ValError::Omit) => {
+                if !extra.exhaustive {
+                    return Err(ValError::Omit);
+                }
+            }
```

Comment:
This is just an example, I think we'd have to implement this in other places, wherever it makes sense.